core/src/main.cpp

```diff
@@ -651,21 +676,38 @@ auto set_log_level(const json& _config) -> void
+			using json_ptr = json::json_pointer;
+
+			if (const auto v = _config.value(json_ptr{_json_ptr_path}, _default_value); !v.empty()) {
+				const auto env_var_upper = boost::to_upper_copy<std::string>(_env_var);
+				log::trace("Setting environment variable [{}] to [{}].", env_var_upper, v);
+				setenv(env_var_upper.c_str(), v.c_str(), 1); // NOLINT(concurrency-mt-unsafe)
+			}
+		};
 
-		if (const auto v = _config.value(json_ptr{_json_ptr_path}, _default_value); !v.empty()) {
-			const auto env_var_upper = boost::to_upper_copy<std::string>(_env_var);
-			log::trace("Setting environment variable [{}] to [{}].", env_var_upper, v);
-			setenv(env_var_upper.c_str(), v.c_str(), 1); // NOLINT(concurrency-mt-unsafe)
-		}
+	const auto set_env_var_int = [&](const auto& _json_ptr_path, const char* _env_var, int _default_value) {
+		using json_ptr = json::json_pointer;
+		const auto v = _config.value(json_ptr{_json_ptr_path}, _default_value);
+		const auto env_var_upper = boost::to_upper_copy<std::string>(_env_var);
+		log::trace("Setting environment variable [{}] to [{}].", env_var_upper, v);
+		const auto v_str = std::to_string(v);
+		setenv(env_var_upper.c_str(), v_str.c_str(), 1); // NOLINT(concurrency-mt-unsafe)
 	};
 
-	set_env_var("/irods_client/tls/client_server_policy", irods::KW_CFG_IRODS_CLIENT_SERVER_POLICY, "CS_NEG_REFUSE");
-	set_env_var("/irods_client/tls/ca_certificate_file", irods::KW_CFG_IRODS_SSL_CA_CERTIFICATE_FILE);
-	set_env_var("/irods_client/tls/certificate_chain_file", irods::KW_CFG_IRODS_SSL_CERTIFICATE_CHAIN_FILE);
-	set_env_var("/irods_client/tls/dh_params_file", irods::KW_CFG_IRODS_SSL_DH_PARAMS_FILE);
-	set_env_var("/irods_client/tls/verify_server", irods::KW_CFG_IRODS_SSL_VERIFY_SERVER, "cert");
+	// clang-format off
+	set_env_var_string("/irods_client/tls/client_server_policy", irods::KW_CFG_IRODS_CLIENT_SERVER_POLICY, "CS_NEG_REFUSE");
+	set_env_var_string("/irods_client/tls/ca_certificate_file", irods::KW_CFG_IRODS_SSL_CA_CERTIFICATE_FILE);
+	set_env_var_string("/irods_client/tls/certificate_chain_file", irods::KW_CFG_IRODS_SSL_CERTIFICATE_CHAIN_FILE);
+	set_env_var_string("/irods_client/tls/dh_params_file", irods::KW_CFG_IRODS_SSL_DH_PARAMS_FILE);
+	set_env_var_string("/irods_client/tls/verify_server", irods::KW_CFG_IRODS_SSL_VERIFY_SERVER, "cert");
+	set_env_var_string("/irods_client/tls/client_server_negotiation", irods::KW_CFG_IRODS_CLIENT_SERVER_NEGOTIATION, "request_server_negotiation");
+	set_env_var_string("/irods_client/tls/encryption_algorithm", irods::KW_CFG_IRODS_ENCRYPTION_ALGORITHM, "AES-256-CBC");
+	set_env_var_int("/irods_client/tls/encryption_key_size", irods::KW_CFG_IRODS_ENCRYPTION_KEY_SIZE, 32);
+	set_env_var_int("/irods_client/tls/encryption_hash_rounds", irods::KW_CFG_IRODS_ENCRYPTION_NUM_HASH_ROUNDS, 16);
+	set_env_var_int("/irods_client/tls/encryption_salt_size", irods::KW_CFG_IRODS_ENCRYPTION_SALT_SIZE, 8);
```

Comment:
Consider having `/irods_client/tls` being a common json pointer and appending as needed.

Might reduce typos if we change this again later.

Reply:
Updated.

Still deciding whether this change is worth it.